src/app/login/services/auth.service.ts

```diff
@@ -1,14 +1,57 @@
 import { Injectable } from '@angular/core';
+import { LoginEvent } from '../models/events';
+import { HttpClient, HttpInterceptor, HttpRequest, HttpHandler, HttpEvent } from '@angular/common/http';
+import { environment } from 'src/environments/environment';
+import { Observable } from 'rxjs';
 
 @Injectable({
   providedIn: 'root'
 })
-export class AuthService {
+export class AuthService implements HttpInterceptor {
```

Comment:
In my opinion we should create a separate AuthInterceptor since we are going to have multiple services using it.

Reply:
As far as I know, you need to only implement HttpInterceptor interface to register your interceptor. I did it this way to encapsulate access token inside AuthService.